src/proposals/20240530/SparkEthereum_20240530.t.sol

```diff
@@ -0,0 +1,179 @@
+        });
+
+        ReserveConfig memory usdcConfigAfter = _findReserveConfigBySymbol(allConfigsAfter, 'USDC');
+        usdcConfigBefore.isSiloed = false;
+        usdcConfigBefore.interestRateStrategy = STABLECOINS_IRM;
+        _validateReserveConfig(usdcConfigBefore, allConfigsAfter);
+        _validateInterestRateStrategy(
+            usdcConfigAfter.interestRateStrategy,
+            usdcConfigAfter.interestRateStrategy,
+            usdIRMValues
+        );
+
+        ReserveConfig memory usdtConfigAfter = _findReserveConfigBySymbol(allConfigsAfter, 'USDT');
+        usdtConfigBefore.isSiloed = false;
+        usdtConfigBefore.interestRateStrategy = STABLECOINS_IRM;
+        _validateReserveConfig(usdtConfigBefore, allConfigsAfter);
+        _validateInterestRateStrategy(
+            usdtConfigAfter.interestRateStrategy,
+            usdtConfigAfter.interestRateStrategy,
+            usdIRMValues
+        );
+
+        ReserveConfig memory wethConfigAfter = _findReserveConfigBySymbol(allConfigsAfter, 'WETH');
+        wethConfigBefore.interestRateStrategy = wethConfigAfter.interestRateStrategy;
+        _validateReserveConfig(wethConfigBefore, allConfigsAfter);
+        _validateInterestRateStrategy(
+            wethConfigAfter.interestRateStrategy,
+            wethConfigAfter.interestRateStrategy,
+            InterestStrategyValues({
+                addressesProvider:             address(poolAddressesProvider),
+                optimalUsageRatio:             wethOldInterestRateStrategy.OPTIMAL_USAGE_RATIO(),
+                optimalStableToTotalDebtRatio: wethOldInterestRateStrategy.OPTIMAL_STABLE_TO_TOTAL_DEBT_RATIO(),
+                baseStableBorrowRate:          0.025e27,
+                stableRateSlope1:              wethOldInterestRateStrategy.getStableRateSlope1(),
+                stableRateSlope2:              wethOldInterestRateStrategy.getStableRateSlope2(),
+                baseVariableBorrowRate:        wethOldInterestRateStrategy.getBaseVariableBorrowRate(),
+                variableRateSlope1:            0.025e27,
+                variableRateSlope2:            wethOldInterestRateStrategy.getVariableRateSlope2()
+            })
+        );
+    }
+
+    function testGnosisSpellExecution() public {
+        executePayload(payload);
+
+        gnosis.selectFork();
+
+        assertEq(IL2BridgeExecutor(Gnosis.AMB_EXECUTOR).getActionsSetCount(), 4);
+
+        gnosis.relayFromHost(true);
+        skip(2 days);
+
+        assertEq(IL2BridgeExecutor(Gnosis.AMB_EXECUTOR).getActionsSetCount(), 5);
+        
+        assertEq(IL2BridgeExecutor(Gnosis.AMB_EXECUTOR).getMinimumDelay(), 8 hours);
+        assertEq(IL2BridgeExecutor(Gnosis.AMB_EXECUTOR).getDelay(),        2 days);
+        IL2BridgeExecutor(Gnosis.AMB_EXECUTOR).execute(4);
+        assertEq(IL2BridgeExecutor(Gnosis.AMB_EXECUTOR).getMinimumDelay(), 0);
+        assertEq(IL2BridgeExecutor(Gnosis.AMB_EXECUTOR).getDelay(),        0);
```

Comment:
If this here is added just to test if the spell was actually executed on gnosis, I'd check the actions set state change for the index 4, before and after execution, instead of selectively doubling part of the the logic tests that are run in full in the gnosis spell tests.

Reply:
The issue is that delegatecall will still succeed even if pointing at an invalid address. You need to check the actual state of the spell. It's a bit awkward, but I don't see how else to add a test to make sure the Gnosis payload is there and properly being executed.